test/e2e/40_tests.bats

```diff
@@ -70,7 +70,27 @@ function setup() {
   [ "$output" = 'False' ]
 }
 
-@test "When test.enable and rollback.enable are set, releases with failed tests are rolled back" {
+@test "When tests fail and test.ignoreFailures is true, release has phase 'TestFailed' and Released condition is True & Tested condition is False" {
+  # Apply the HelmRelease that has test failure
+  kubectl apply -f "$FIXTURES_DIR/releases/test/fail-ignored.yaml" >&3
+
+  # Wait for test failure
+  poll_until_true 'test failure' "kubectl -n $E2E_NAMESPACE logs deploy/helm-operator | grep -E \"test failed\""
+
+  # Assert Helm Release `Phase` is `TestFailed`
+  poll_until_equals 'release phase is TestFailed' 'TestFailed' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o jsonpath='{.status.phase}'"
+
+  # Assert `Tested` condition is `False`
+  poll_until_equals 'tested condition False' 'False' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o jsonpath='{.status.conditions[?(@.type==\"Tested\")].status}'"
+
+  # Assert test failures are ignored but logged
+  poll_until_true 'test failure - ignore failures' "kubectl -n $E2E_NAMESPACE logs deploy/helm-operator | grep -E \"test failed - ignoring failures\""
+
+  # Assert `Released` condition is `True`
+  poll_until_equals 'released condition True' 'True' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o jsonpath='{.status.conditions[?(@.type==\"Released\")].status}'"
+}
```

Comment:
Can you add a check that `status.revision` is set correctly to the successful (despite ignored failed test) chart revision?

Reply:
@seaneagan do you mean:
`poll_until_equals 'revision match' "4.0.1" "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o jsonpath='{.status.revision}'"`

status.revision should be the chart.version if it's not a git sync'd chart? If you mean asserting on the revision '4.0.1' I can add it but I'm not sure how much value it adds?